### DISCUSSION TITLE
Support for --stdinpass or DEPnotify/jamfhelper during just download?

### DISCUSSION THREAD

**Discussion by warmsoda** [grahampugh/erase-install#213] [Q&A]
Hey everyone, 3 quick questions...

1. I'm pretty sure the answer is no (without significant editing) but for our enrolment/update workflow we use `--stdinpass` with `startosinstall` to avoid the password prompt (additionally, none of our users are admins) and I was hoping there is some support or way I can add `--stdinpass` into the fold.

2. (kind of related) I've gotten the '--reinstall' process to show DEPnotify messages without issue, but I've been using just the 'download' portion (example below) and then scripting out the `startosinstall` afterwards for Apple Silicon macs using the `--stdinpass` proccess; this leads to no DEPnotify message/GUI. I assumed afterwards this was supposed this way, but I'm just double-checking, is there any way to get the DEPnotify (or jamf helper) to show for just the download portion?

3. Is there any integrated way for erase-install to delete DEPnotifty as part of the `--cleanup-after-use`?
Download and Reinstall - DEPnotify messages shown
`/Library/Management/erase-install/erase-install.sh --sameos --pkg --reinstall --depnotify --cleanup-after-use --check-power --power-wait-limit 180`

Download - **No** DEPnotify messages shown
`/Library/Management/erase-install/erase-install.sh --sameos --pkg --move --depnotify --check-power --power-wait-limit 180`

**Comment by grahampugh** [comment]
I remain resistant to add functionality to supply the volume owner's password in plain text. 

No, I didn't include the deletion of DEPNotify. There were too many caveats there, like if you were using it for other purposes. And it's hidden away in Utilities, small, and does nothing on its own, so I considered it harmless to remain.

**Reply by warmsoda** [reply]
Appreciate the quick reply!

Any thoughts on 2? Is there a way to get the **non** `--reinstall` workflow to show DEP/jamf messages?

**Reply by grahampugh** [reply]
No, no way to do that. I didn't see the need for that since all you're doing is downloading an app.
